File: go/process/tiny.go
// tiny.go
// minimal program to observe VSZ vs RSS in Go.
// the Go runtime pre-maps a large virtual address region for its GC heap,
// so VSZ will be ~1 GB while RSS is only a few MB.
//
// usage:
//   go build -o tiny tiny.go
//   ./tiny &
//   ps -o pid,vsz,rss,comm -p $!
//   kill $!
package main

import (
	"fmt"
	"time"
)

func main() {
	fmt.Println("hello")
	time.Sleep(30 * time.Second)
}

File: rust/process/tiny.rs
// tiny.rs
// minimal program to observe VSZ vs RSS in Rust.
// no GC, no runtime heap pre-mapping -- VSZ will be ~4-8 MB, RSS ~2-3 MB.
// compare with the Go version to see the difference.
//
// usage:
//   rustc -o tiny_rs tiny.rs
//   ./tiny_rs &
//   ps -o pid,vsz,rss,comm -p $!
//   kill $!
use std::{thread, time::Duration};

fn main() {
    println!("hello");
    thread::sleep(Duration::from_secs(30));
}
